src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/ProjectSystemCache.cs

```diff
@@ -318,26 +318,31 @@ public bool TryGetProjectNameByShortName(string name, out ProjectNames projectNa
                 throw new ArgumentException(CommonResources.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
             }
 
-            projectNames = null;
-
             _readerWriterLock.EnterReadLock();
 
             try
             {
-                HashSet<ProjectNames> values;
-                if (_shortNameCache.TryGetValue(name, out values))
-                {
-                    // Get the item at the front of the queue
-                    projectNames = values.Count == 1 ? values.Single() : null;
-
-                    // Only return true if the short name is unambiguous
-                    return projectNames != null;
-                }
+                return TryGetProjectNameByShortNameWithoutLock(name, out projectNames);
             }
             finally
             {
                 _readerWriterLock.ExitReadLock();
             }
+        }
+
+        private bool TryGetProjectNameByShortNameWithoutLock(string name, out ProjectNames projectNames)
+        {
+            projectNames = null;
+
+            HashSet<ProjectNames> values;
+            if (_shortNameCache.TryGetValue(name, out values))
+            {
+                // Get the item at the front of the queue
```

Comment:
This comment could be improved, from the tests it looks like this only returns if there are no ambiguous conflicts

Looks like it was this way before also